lit_gpt/lora.py

```diff
@@ -87,6 +88,71 @@ def __init__(self, r: int, lora_alpha: int, lora_dropout: float, merge_weights:
         self.merge_weights = merge_weights
 
 
+class Linear(nn.Linear, LoRALayer):
+    # LoRA implemented in a dense layer
+    def __init__(
+        self,
+        in_features: int,
+        out_features: int,
+        r: int = 0,
+        lora_alpha: int = 1,
+        lora_dropout: float = 0.,
+        fan_in_fan_out: bool = False, # Set this to True if the layer to replace stores weight like (fan_in, fan_out)
+        merge_weights: bool = True,
+        **kwargs
+    ):
+        nn.Linear.__init__(self, in_features, out_features, **kwargs)
```

Comment:
```suggestion
        super().__init__(in_features, out_features, **kwargs)
```

Reply:
Could you explain your reasoning? Because I'm not sure about it.
`Linear` class inherits two classes: `nn.Linear` and `LoRALayer`. 
So currently the code explicitly provides arguments for each parent class separately.

From the docs (or from this [stackoverflow](https://stackoverflow.com/questions/3277367/how-does-pythons-super-work-with-multiple-inheritance) post) I've found that `super().__init__` in case of multiple inheritance will first call init method of the first parent class and, if the first parent class doesn't expect such arguments, calls the second parent class with leftover args. Basically calling init method of parent classes as they are listed from left to right.

So there are multiple issues:
1. some arguments from `**kwargs` might go into the next parent classes if they weren't used in the current class
2. I don't know what is gonna happen when `super` will call the second parent class because of unused arguments and then the code calls this class on more time
3. if we have `super` for the `nn.Linear` then why we leave `LoRALayer` as is?

In theory we can have one big `super` method where arguments for both classes are listed and let python to deal with it, but I kinda feel uncomfortable about this.

----
Or maybe I'm digging too deep or simply wrong 🤷‍♂️ .

Reply:
The motivation for this suggestion is that we are patching `nn.Linear` at runtime and that can break it: https://github.com/Lightning-AI/lit-gpt/commit/ddfcbfb301e4bcec37be0a0a36edfeda12984e9b

Reply:
Ok. That makes sense. 
I've found one more way how you can call init method of the first parent class not by name - by position:
```python
self.__class__.__bases__[0].__init__(...
```
But be careful with this knowledge: use it in emergency only. After introducing such "quality" of code to this repo - half of github starts might suddenly disappear.

Reply:
Haha! Then let's go with `super().__init__`? I don't think the `**kwargs` issue is relevant to the code today. If it becomes a problem we can revisit it